### PR TITLE
AI Upgrade System Code Holding PR

### PR DESCRIPTION
**Pull request by Dorsisdwarf**
This is not a PR to be merged, as can be seen from the super-fucked up commit log /diff. It's just putting the code up where people can see/steal it, because I didnt finish on time and am now too busy to complete the feature

### CODE REVIEW

**Comment by Jalleo** [comment]
I see what you mean wow those commits went mad

File: code/modules/mob/living/silicon/ai/ai.dm [hunk 1]
```diff
@@ -31,13 +31,21 @@ var/list/ai_list = list()
...
 	var/obj/item/device/multitool/aiMulti = null
 	var/obj/machinery/bot/Bot
 	var/tracking = 0 //this is 1 if the AI is currently tracking somebody, but the track has not yet been completed.
 
+
+	//RESEARCH
+	var/researchrate = 1
+	var/researchtime
+	var/researching = 0
+	var/list/subsystems = list("doors"=1,"lights"=1,"comms"=1) //shows subsystems AI can use
```
Comment (RemieRichards) [inline code/modules/mob/living/silicon/ai/ai.dm]: Not strings, No.

File: code/modules/mob/living/silicon/ai/ai.dm [hunk 2]
```diff
@@ -31,13 +31,21 @@ var/list/ai_list = list()
...
+	var/list/researches = list("bot interface","civilian radio","command/security radio","vox")
```
Comment (RemieRichards) [inline code/modules/mob/living/silicon/ai/ai.dm]: Ditto

File: code/modules/mob/living/silicon/ai/upgrades.dm [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+		return 0 //need to make this give the upgrade back
+
+	researching = 0
+	subsystems |= upgrade
+	subsystems[upgrade] = 1
+	researches -= upgrade //needs to check upgrade is /in/ researches first
+	src << "Systems upgraded. The [upgrade] subsystem is now active."
+
+/mob/living/silicon/ai/proc/handleresearchrate(var/syntax = 0)
+	if(syntax)
```
Comment (RemieRichards) [inline code/modules/mob/living/silicon/ai/upgrades.dm]: shit argument name.

File: code/modules/mob/living/silicon/ai/upgrades.dm [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+		researchrate += 1
```
Comment (RemieRichards) [inline code/modules/mob/living/silicon/ai/upgrades.dm]: ```++```

File: code/modules/mob/living/silicon/ai/upgrades.dm [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+		return
+	else
+		researchrate -= 1
```
Comment (RemieRichards) [inline code/modules/mob/living/silicon/ai/upgrades.dm]: ```--``` instead of ```-= 1```

**Comment by RemieRichards** [comment]
You've messed up a lot of config/text files.

**Comment by xxalpha** [comment]
>6000+ 5000-

What sort of unholyness is this

**Comment by Dorsisdwarf** [comment]
Line endings + drive corruption. anyone using this shit should probbably just type it into their byond themself, because this is FUCKED (hence abandonment)

File: code/modules/mob/living/silicon/ai/ai.dm (hunk 1 above)
Comment (Dorsisdwarf) [inline code/modules/mob/living/silicon/ai/ai.dm]: What's wrong with this, why do headset keys do this if it's bad, and what would be a good solution?

Comment (RemieRichards) [inline code/modules/mob/living/silicon/ai/ai.dm]: "It's done elsewhere in the code" != "It's acceptable"

You should be using ```#defines```, yes they'll be strings in the back end, but it prevents typos.

**Comment by Shadowlight213** [comment]
So what exactly does it do?

**Comment by Razharas** [comment]
@Dorsisdwarf 
What is the purpose of this pr?
Do you want this to be merged?

**Comment by Dorsisdwarf** [comment]
Closed before raz can yolomerge and fuck EVERYTHING EVER UP

**Comment by Jalleo** [comment]
It was a PR for anyone to rip the code from to finish razh

**Pull request closed without merging**